core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -166,67 +170,65 @@ private[storage] class BlockManagerDecommissioner(
   private val migrationPeers =
     mutable.HashMap[BlockManagerId, ShuffleMigrationRunnable]()
 
-  private lazy val rddBlockMigrationExecutor =
-    ThreadUtils.newDaemonSingleThreadExecutor("block-manager-decommission-rdd")
+  private val rddBlockMigrationExecutor =
```

Comment:
I don't understand what this refactoring gives us.

Reply:
IIUC,  it's decorated with `lazy` because we won't initiate it when rdd migration disabled. However, it isn't implemented correctly as we always access it in `stop()`, which breaks our expectation (I think). This refactor fixes the problem.

Reply:
Makes sense.